faceV4.py

```diff
@@ -0,0 +1,202 @@
+        today = datetime.now().replace(microsecond=0, second=0, minute=0)
+        end_time = today
+        start_time =  (end_time - timedelta(days=365))
+        # any parameters
+        payload = {
+        }
+        sess.headers = {
+            "x-auth-scheme": "api-token",
+            "x-auth-apikey": api_key
+        }
+        resp = sess.post(endpoint, json=payload,
+        verify=False)
+        content = resp.content
+        data = json.loads(content)
+        return data
+
+#gets the camera_name from the uuid
+def camera_name(uuid, data):
+    for value in data['cameraStates']:
+        if uuid == value['uuid']:
+            return value['name']
+
+#uses the api to get the recent faces
+def recent_faces():
+    if __name__ == "__main__":
+        # url of the api
+        endpoint = "https://api2.rhombussystems.com/api/face/getRecentFaceEventsV2"
+        api_key = args.APIkey
+        sess = requests.session()
+        today = datetime.now().replace(microsecond=0, second=0, minute=0)
+        end_time = today
+        start_time =  (end_time - timedelta(days=365))
+        #checks to see if the user put in a start and end time if not it defaults to one hour before current
+        if args.startTime:
+            start = milliseconds_time(args.startTime)
+        else:
+            start = int(round((time.time() - 3600) * 1000))
+        if args.endTime:
+            end = milliseconds_time(args.endTime)
+        else:
+            end = int(round(time.time() * 1000))
+        #any parameters
+        payload = {
+            "filter":{"types":[args.filter]}, #arguement to filter alert, trusted, named, or other
+            "interval":{
+                "end": end,
+                "start": start
+            }
+        }
+        sess.headers = {
+            "x-auth-scheme": "api-token",
+            "x-auth-apikey": api_key
+        }
+        resp = sess.post(endpoint, json=payload,
+        verify=False)
+        content = resp.content
+        data = json.loads(content)
+        return data
+
+#adds the name, timestamp, camera name, and sighting number to a csv
+def csv_add(value, csv_data, count, data_camera, baseURL, header):
+    timestamp = human_time(value['eventTimestamp'])
+    csv_data.append([])
+    name = value['faceName']
+    csv_data[count].append(name)
+    csv_data[count].append(timestamp)
+    camera = camera_name(value['deviceUuid'], data_camera)
+    csv_data[count].append(camera)
+    thumbnail = baseURL + value['thumbnailS3Key']
+    saving_img(thumbnail, name, count)
+    csv_data[count].append(count + 1)
+    with open(args.report + '/' + args.csv + '.csv', 'w', newline = '') as f:
+        writer = csv.writer(f)     # create the csv writer
+        writer.writerow(header)    # write the header
+        writer.writerows(csv_data) # write the data
+
+def csv_work(data_recentFaces):
+    count = 0
+    baseURL = "https://media.rhombussystems.com/media/faces?s3ObjectKey="
```

Comment:
rename to `mediaBaseURL` or something more specific - we don't want to confuse this url with the api urls.